habitat_baselines/common/utils.py

```diff
@@ -174,3 +222,111 @@ def generate_video(
+    """Resizes an img so that the shortest side is length of size.
+
+    Args:
+        img: the array object that needs to be resized (HWC) or (NHWC)
+        size: the size that you want the shortest edge to be resize to
+        channels_first: a boolean that specifies the img is (CHW) or (NCHW)
+    Returns:
+        The resized array as a torch tensor.
+    """
+    no_batch_dim = len(img.shape) == 3
+    if len(img.shape) < 3 or len(img.shape) > 5:
+        raise NotImplementedError()
+    img = _to_tensor(img)
+    if no_batch_dim:
+        img = img.unsqueeze(0)  # Adds a batch dimension
+    if channels_first:
+        # NCHW
+        h, w = img.shape[-2:]
+    else:
+        # NHWC
+        h, w = img.shape[-3:-1]
+        if len(img.shape) == 4:
+            img = img.permute(0, 3, 1, 2)
+        else:
+            img = img.permute(0, 1, 4, 2, 3)
+
+    if w > h:
+        percent = size / h
+    else:
+        percent = size / w
+    h *= percent
+    w *= percent
+    h = int(h)
+    w = int(w)
+    img = torch.nn.functional.interpolate(
+        img.float(), size=(h, w), mode="area"
+    ).to(dtype=img.dtype)
+    if not channels_first:
+        if len(img.shape) == 4:
+            img = img.permute(0, 2, 3, 1)
+        else:
+            img = img.permute(0, 1, 3, 4, 2)
+    if no_batch_dim:
+        img = img.squeeze(dim=0)  # Removes the batch dimension
+    return img
+
+
+def center_crop(img, cropx: int, cropy: int, channels_first: bool = False):
+    """Performs a center crop on an image.
+
+    Args:
+        img: the array object that needs to be resized (either batched or unbatched)
+        size: the size that you want the shortest edge to be resize to
+        channels_first: If it's in NCHW
+    Returns:
+        the resized array
+    """
+    if channels_first:
+        # NCHW
+        y, x = img.shape[-2:]
+    else:
+        # NHWC
+        y, x = img.shape[-3:-1]
+
+    startx = x // 2 - (cropx // 2)
+    starty = y // 2 - (cropy // 2)
+    if channels_first:
+        return img[..., starty : starty + cropy, startx : startx + cropx]
+    else:
+        return img[..., starty : starty + cropy, startx : startx + cropx, :]
+
+
+def apply_ppo_data_augs(
+    observations: Dict[str, Any],
+    resize: int,
+    center_crop_size: int,
+    channels_first: bool = False,
+) -> Dict[str, Any]:
+    for k, obs in observations.items():
+        if k in ["rgb", "depth", "semantic"]:
+            if resize != 0:
+                obs = image_resize_shortest_edge(
+                    obs, resize, channels_first=channels_first
+                )
+            if center_crop_size != 0:
+                obs = center_crop(
+                    obs,
+                    center_crop_size,
+                    center_crop_size,
+                    channels_first=channels_first,
+                )
+            observations[k] = obs
+    return observations
+
+
+def overwrite_gym_box(box: Box, shape: tuple) -> Box:
```

Comment:
Maybe `overwrite_gym_box_height_width`.

Reply:
It can also overwrite channels if specified though or any other aspects of shape.